addons/web/static/src/scss/ui.scss

```diff
@@ -183,3 +183,14 @@ span.o_force_ltr {
         }
     }
 }
+
+.o_has_ui_app_icon {
```

Comment:
I don't think that we need a specific class.
If its only purpose is to select the `.o_ui_app_icon` parent, we can easily retrieve it in plan CSS


```scss
.o_ui_app_icon {
   width: var(--oi-font-size, 1em);
   aspect-ratio: 1;
   
   :not(:hover) > & {
       --oi-color: currentColor;
   }
}
```